lib/pure/math.nim

```diff
@@ -949,18 +949,25 @@ func frexp*[T: float32|float64](x: T): tuple[frac: T, exp: int] {.inline.} =
     doAssert frexp(8.0) == (0.5, 4)
     doAssert frexp(-8.0) == (-0.5, 4)
     doAssert frexp(0.0) == (0.0, 0)
+
     # special cases:
-    when not defined(windows):
-      doAssert frexp(-0.0) == (-0.0, 0) # signbit preserved for +-0
-      doAssert frexp(Inf).frac == Inf # +- Inf preserved
-      doAssert frexp(NaN).frac.isNaN
+    doAssert frexp(-0.0) == (-0.0, 0) # signbit preserved for +-0
+    doAssert frexp(Inf).frac == Inf # +- Inf preserved
+    doAssert frexp(NaN).frac.isNaN
+
   when not defined(js):
     var exp: cint
     result.frac = c_frexp2(x, exp)
     result.exp = exp
   else:
     if x == 0.0:
-      result = (0.0, 0)
+      # reuse signbit implementation
+      let uintBuffer = toBitsImpl(x)
+      if (uintBuffer[1] shr 31) != 0:
+        # x is -0.0
```

Comment:
```suggestion
      if x.signbit:
```

Reply:
The problem is that `signbit` is `since (1, 5, 1)`, see also https://github.com/nim-lang/Nim/pull/16725#discussion_r568301079.